website/source/docs/providers/postgresql/r/postgresql_schema.html.markdown

```diff
@@ -8,37 +8,86 @@ description: |-
 
 # postgresql\_schema
 
-The ``postgresql_schema`` resource creates and manages a schema within a
-PostgreSQL database.
+The ``postgresql_schema`` resource creates and manages a [schema
+objects](https://www.postgresql.org/docs/current/static/ddl-schemas.html) within
+a PostgreSQL database.
 
 
 ## Usage
 
 ```
+resource "postgresql_role" "app_www" {
+  name = "app_www"
+}
+
+resource "postgresql_role" "app_dba" {
+  name = "app_dba"
+}
+
+resource "postgresql_role" "app_releng" {
+  name = "app_releng"
+}
+
 resource "postgresql_schema" "my_schema" {
-  name = "my_schema"
+  name  = "my_schema"
+  owner = "postgres"
+
+  policy {
+    usage = true
+    role = "${postgresql_role.app_www.name}"
+  }
+
+  # app_releng can create new objects in the schema.  This is the role that
+  # migrations are executed as.
+  policy {
+    create = true
+    usage = true
+    role = "${postgresql_role.app_releng.name}"
+  }
+
+  policy {
+    create_with_grant = true
+    usage_with_grant = true
+    role = "${postgresql_role.app_dba.name}"
+  }
 }
 ```
 
 ## Argument Reference
 
 * `name` - (Required) The name of the schema. Must be unique in the PostgreSQL
   database instance where it is configured.
+* `owner` - (Optional) The ROLE who owns the schema.
+* `policy` - (Optional) Can be specified multiple times for each policy.  Each
+    policy block supports fields documented below.
+
+The `policy` block supports:
+
+* `create` - (Optional) Should the specified ROLE have CREATE privileges to the specified SCHEMA.
+* `create_with_grant` - (Optional) Should the specified ROLE have CREATE privileges to the specified SCHEMA and the ability to GRANT the CREATE privilege to other ROLEs.
+* `role` - (Optional) The ROLE who is receiving the policy.  If this value is empty or not specified it implies the policy is referring to the [`PUBLIC` role](https://www.postgresql.org/docs/current/static/sql-grant.html).
+* `usage` - (Optional) Should the specified ROLE have USAGE privileges to the specified SCHEMA.
+* `usage_with_grant` - (Optional) Should the specified ROLE have USAGE privileges to the specified SCHEMA and the ability to GRANT the USAGE privilege to other ROLEs.
+
+~> **NOTE on `policy`:** The permissions of a role specified in multiple policy blocks is cumulative.  For example, if the same role is specified in two different `policy` each with different permissions (e.g. `create` and `usage_with_grant`, respectively), then the specified role with have both `create` and `usage_with_grant` privileges.
 
 ## Import Example
 
 `postgresql_schema` supports importing resources.  Supposing the following
```

Comment:
I started there but added the examples because the import without a config is opaque to new users who want to move to TF.  I can remove the import sample if you'd like but think it's a net win for users for the time being.  Happy to go either way, however.